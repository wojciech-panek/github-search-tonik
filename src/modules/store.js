import { createStore, applyMiddleware, compose } from 'redux';
import Immutable from 'seamless-immutable';
import createSagaMiddleware from 'redux-saga';
import createReducer from './reducers';
import rootSaga from './sagas';

const sagaMiddleware = createSagaMiddleware();

export default function configureStore(initialState = {}) {
  const middlewares = [sagaMiddleware];

  const store = createStore(createReducer(), Immutable(initialState), compose(applyMiddleware(...middlewares)));

  sagaMiddleware.run(rootSaga);

  return store;
}
