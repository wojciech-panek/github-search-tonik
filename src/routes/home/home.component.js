import React from 'react';

import { Container } from './home.styles';

export const Home = () => {
  return <Container>Home</Container>;
};
