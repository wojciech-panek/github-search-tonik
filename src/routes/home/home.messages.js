/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  pageTitle: {
    id: 'home.pageTitle',
    defaultMessage: 'Home',
  },
  welcome: {
    id: 'home.welcome',
    defaultMessage: 'Home',
  },
});
