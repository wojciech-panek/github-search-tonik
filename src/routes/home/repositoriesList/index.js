export { RepositoriesList } from './repositoriesList.component';
