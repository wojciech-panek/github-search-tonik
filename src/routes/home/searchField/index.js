export { SearchField } from './searchField.component';
