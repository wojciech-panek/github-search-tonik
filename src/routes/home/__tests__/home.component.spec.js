import React from 'react';
import { shallow } from 'enzyme';
import { clone } from 'ramda';

import { Home } from '../home.component';
import { store } from '../../../fixtures/store';

let mockStore = clone(store);
jest.mock('react-redux', () => ({
  useSelector: selector => selector(mockStore),
}));

describe('Home: Component', () => {
  const defaultProps = {};

  const component = props => <Home {...defaultProps} {...props} />;

  it('should render correctly', () => {
    const wrapper = shallow(component());
    expect(wrapper).toMatchSnapshot();
  });
});
