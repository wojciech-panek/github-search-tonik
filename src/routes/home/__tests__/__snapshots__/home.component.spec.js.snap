// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home: Component should render correctly 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="home.pageTitle / Home / {}"
  />
  <FormattedMessage
    defaultMessage="Home"
    id="home.welcome"
  />
</styled.div>
`;
